ios/browser/api/ipfs/ipfs_api.mm

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/ipfs/ipfs_api.h"
+
+#include "brave/components/ipfs/ipfs_utils.h"
+#include "components/user_prefs/user_prefs.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state_manager.h"
+#import "net/base/mac/url_conversions.h"
+#include "url/gurl.h"
+
+@implementation IpfsAPI {
+  ChromeBrowserState* _mainBrowserState;  // NOT OWNED
+}
+
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)mainBrowserState {
+  if ((self = [super init])) {
+    _mainBrowserState = mainBrowserState;
+  }
+  return self;
+}
+
+- (NSURL*)resolveGatewayUrlFor:(NSString*)input {
+  auto* browserState = _mainBrowserState;
+  GURL input_gurl = net::GURLWithNSURL([NSURL URLWithString:input]);
+  if (!input_gurl.is_valid() && !input_gurl.SchemeIs(ipfs::kIpfsScheme)) {
+    return nullptr;
+  }
+  GURL output;
+  PrefService* prefs = user_prefs::UserPrefs::Get(browserState);
```

Comment:
Use `_mainBrowserState` directly instead of creating a local variable?
Same for the three functions below.